nwb_conversion_tools/datainterfaces/behavior/deeplabcut/deeplabcutdatainterface.py

```diff
@@ -0,0 +1,143 @@
+"""Authors: Saksham Sharda, Cody Baker, Ben Dichter."""
+import datetime
+import os
+import warnings
+
+from ....basedatainterface import BaseDataInterface
+from ....tools.nwb_helpers import get_module
+
+try:
+    from dlc2nwb.utils import (
+        get_movie_timestamps,
+        _ensure_individuals_in_header,
+        __version__,
+        auxiliaryfunctions,
+        VideoReader,
+        PoseEstimationSeries,
+        PoseEstimation,
+        pd,
+        pickle,
+    )
+
+    HAVE_DLC2NWB = True
+except ImportError:
+    HAVE_DLC2NWB = False
+
+
+class DeepLabCutInterface(BaseDataInterface):
+    """Data interface for DeepLabCut datasets"""
+
+    def __init__(self, dlc_file_path, config_file_path):
+        """
+        Interface for writing DLC's h5 files to nwb using dlc2nwb.
+
+        Parameters
+        ----------
+        dlc_file_path: FilePathType
+            path to the h5 file output by dlc.
+        config_file_path: FilePathType
+            path to .yml config file
+        """
+        if "DLC" not in dlc_file_path or not dlc_file_path.endswith(".h5"):
+            raise IOError("The file passed in is not a DeepLabCut h5 data file.")
+        assert HAVE_DLC2NWB, "to use this datainterface: 'pip install dlc2nwb'"
+        self._config_file = auxiliaryfunctions.read_config(config_file_path)
+        super().__init__(dlc_file_path=dlc_file_path, config_file_path=config_file_path)
+        self._derived_metadata = self._metadata_from_config()
+
+    def _metadata_from_config(self):
+        dlc_file_path = self.source_data["dlc_file_path"]
+        vidname, scorer = os.path.split(dlc_file_path)[-1].split("DLC")
+        scorer = "DLC" + os.path.splitext(scorer)[0]
+        return dict(vidname=vidname, scorer=scorer)
+
+    def get_metadata(self):
+        metadata = dict(
+            NWBFile=dict(
+                session_description=self._config_file["Task"],
+                experimenter=self._config_file["scorer"],
```

Comment:
I also feel like this, out of all the interfaces we support, ought to be the first one where we actually require (and can hopefully extract) `Subject` info, since if someone is using DLC at all it would imply there is a Subject being tracked.